app/src/androidTest/java/sw19/moring03/paint/SaveEspressoTest.java

```diff
@@ -27,17 +31,29 @@
 
     @Test
     public void testSaveButtonExists() {
-        onView(withId(R.id.saveButton)).check(matches(isDisplayed()));
+        try {
```

Comment:
could you please explain why the nonmatchingviewexception should be happening here?

it should be impossible that the view does not exist since you navigate through the app with espresso tests.
if it can throw such an exception add "throws NoMatchingViewException" to the method and don't handle it internally

if you reallly really need that try catch than at least put the duplicate code of those two cases after the try catch block